src/imars3d/backend/workflow/engine.py

```diff
@@ -194,6 +197,8 @@ def run(self) -> None:
             peek = self._instrospect_task_function(task["function"])
             inputs = self._resolve_inputs(task.get("inputs", {}), peek.paramdict)
             outputs = peek.function(**inputs)
+            if type(outputs) is not tuple:
```

Comment:
[This violates the schema](https://github.com/ornlneutronimaging/iMars3D/blob/next/src/imars3d/backend/workflow/schema.json#L51). Function outputs **must** be an iterable (preferably a list) and not a string. If there's a function returning something else than an iterable, the source code must be changed

Reply:
Functions that have multiple returns return a tuple naturally, the case this is specifically for is when it just returns an array.  the way we interpret function_outputs later mistakes the contents of the array as the individual outputs, thus if we return an array of size 600(data) from denoise and only expect one function output, it fails the check.  I think it would be better to catch it here and let functions operate as normal functions without having to worry about wrapping their output, that may be error prone otherwise.

Reply:
How about we require that the _function outputs_ are either `None`, `list`, or `tuple`?
```python
    def _validate_outputs(task_outputs: TaksOutputTypes,
                          function_outputs: Optional[TaksOutputTypes] = None) -> None:
  
        def validate_type(outputs):
            if not isinstance(outputs, (list, tuple)):
                raise WorkflowEngineError("Task and Function outputs must be a list or a tuple")

        validate_type(task_outputs)
        if function_outputs:
            validate_type(function_outputs)
            if len(task_outputs) != len(function_outputs):
                error = "Task and Function have different number of outputs"
                raise WorkflowEngineError(error)
```

Reply:
Is there a reason we cant just handle non-tuples gracefully instead of erroring out? It would be easier to maintain. and as far as the json schema is concerned it would remain the same as tuples are serialized as arrays.

none of our functions return a tuple and expect it to be interpreted as a lone output, and I struggle to think of a usecase where we would.